### ISSUE TITLE
Needs `-latomic` on RPi

### ISSUE THREAD

**Issue by Kernald** [badaix/snapcast#255]
Building Snapcast on a first-gen Raspberry Pi (running an up-to-date Archlinux), the build ends up in undefined references:
```
g++ -march=armv6 -mfloat-abi=hard -mfpu=vfp -O2 -pipe -fstack-protector --param=ssp-buffer-size=4  -std=c++0x -Wall -Wno-unused-function -O3 -DASIO_STANDALONE -DVERSION=\"0.11.1\" -I. -I.. -isystem ../externals/asio/asio/include -I../externals/popl/include -pthread -DHAS_OGG -DHAS_ALSA -DHAS_AVAHI -DHAS_DAEMON -o snapclient snapClient.o stream.o clientConnection.o timeProvider.o player/player.o decoder/pcmDecoder.o decoder/oggDecoder.o decoder/flacDecoder.o controller.o ../message/pcmChunk.o ../common/log.o ../common/sampleFormat.o player/alsaPlayer.o browseZeroConf/browseAvahi.o -lrt -lasound -logg -lvorbis -lFLAC -lavahi-client -lavahi-common -static-libgcc -static-libstdc++
controller.o: In function `Controller::worker()':
controller.cpp:(.text+0x19b8): undefined reference to `__atomic_load_8'
stream.o: In function `Stream::getPlayerChunk(void*, std::chrono::duration<long long, std::ratio<1ll, 1000000ll> > const&, unsigned long)':
stream.cpp:(.text+0x28bc): undefined reference to `__atomic_load_8'
stream.cpp:(.text+0x2ae0): undefined reference to `__atomic_load_8'
stream.cpp:(.text+0x3780): undefined reference to `__atomic_load_8'
stream.cpp:(.text+0x4244): undefined reference to `__atomic_load_8'
timeProvider.o: In function `TimeProvider::setDiffToServer(double)':
timeProvider.cpp:(.text+0x21c): undefined reference to `__atomic_store_8'
timeProvider.cpp:(.text+0x304): undefined reference to `__atomic_store_8'
timeProvider.o: In function `TimeProvider::setDiff(tv const&, tv const&)':
timeProvider.cpp:(.text+0x638): undefined reference to `__atomic_store_8'
timeProvider.cpp:(.text+0x718): undefined reference to `__atomic_store_8'
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:70: snapclient] Error 1
make[1]: Leaving directory '/tmp/packerbuild-1001/snapcast/snapcast/src/snapcast/client'
```

Linking to `libatomic` works fine. The link stage ends up being:
```
g++ -march=armv6 -mfloat-abi=hard -mfpu=vfp -O2 -pipe -fstack-protector --param=ssp-buffer-size=4  -std=c++0x -Wall -Wno-unused-function -O3 -DASIO_STANDALONE -DVERSION=\"0.11.1\" -I. -I.. -isystem ../externals/asio/asio/include -I../externals/popl/include -pthread -DHAS_OGG -DHAS_ALSA -DHAS_AVAHI -DHAS_DAEMON -o snapclient snapClient.o stream.o clientConnection.o timeProvider.o player/player.o decoder/pcmDecoder.o decoder/oggDecoder.o decoder/flacDecoder.o controller.o ../message/pcmChunk.o ../common/log.o ../common/sampleFormat.o player/alsaPlayer.o browseZeroConf/browseAvahi.o -lrt -lasound -logg -lvorbis -lFLAC -lavahi-client -lavahi-common -static-libgcc -static-libstdc++ -latomic
```

This seems to affect the client only.

**Comment by mokman** [comment]
i try to install snapchat from aur on rpi zero and get the error above. is there a workaround? i do not understand what linking to libatomic means.
thanks

**Comment by Kernald** [comment]
You'll have to edit the client Makefile (`client/Makefile`), more specifically this line: https://github.com/badaix/snapcast/blob/master/client/Makefile#L58 and add `-latomic`. This becomes:
```
LDFLAGS = -lrt -lasound -logg -lvorbis -lFLAC -lavahi-client -lavahi-common -static-libgcc -static-libstdc++
```

**Comment by slackline** [comment]
Thank you for the cross reference, hadn't found this when searching.

**Comment by badaix** [comment]
Fixed in version [v0.12.0](https://github.com/badaix/snapcast/releases/tag/v0.12.0)
